src/Caching/Hybrid/src/Internal/DefaultHybridCache.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Caching.Hybrid.Internal;
+
+/// <summary>
+/// The inbuilt ASP.NET implementation of <see cref="HybridCache"/>
+/// </summary>
+internal sealed class DefaultHybridCache : HybridCache
+{
+    private readonly IDistributedCache backendCache;
+    private readonly IServiceProvider services;
+    private readonly HybridCacheOptions options;
+    public DefaultHybridCache(IOptions<HybridCacheOptions> options, IDistributedCache backendCache, IServiceProvider services)
+    {
+        this.backendCache = backendCache ?? throw new ArgumentNullException(nameof(backendCache));
+        this.services = services ?? throw new ArgumentNullException(nameof(services));
+        this.options = options.Value;
+    }
+    internal HybridCacheOptions Options => options;
+
+    public override ValueTask<T> GetOrCreateAsync<TState, T>(string key, TState state, Func<TState, CancellationToken, ValueTask<T>> underlyingDataCallback, HybridCacheEntryOptions? options = null, ICollection<string>? tags = null, CancellationToken token = default)
```

Comment:
I think the state parameter usually comes after the delegate parameter.

Reply:
mixture; `ThreadPool`: callback then state; `ILogger`: state then callback

Reply:
argument for state first: the state is the thing that is most likely to change on a per-call basis, where-as the callback here is probably a static lambda; because of that, the argument value is more eyeballs-worthy for the reader to understand what the code is doing when reading it later

Reply:
If we're not consistent, I also prefer state-first.